### ISSUE TITLE
Not all punycode domains are parsed correctly

### ISSUE THREAD

**Issue by ruurtjan** [lemonlabsuk/scala-uri#424]
Thanks for maintaining Scala-URI! I've been using it for [NsLookup.io](https://www.nslookup.io) for a while now :)

**Describe the bug**
This domain exists and can be visited in a webbrowser, but fails to parse correctly: [http://👸🏽.cf](http://👸🏽.cf). 

**To Reproduce**
```scala
object ReproducePunycodeBug extends App {
  import io.lemonlabs.uri.Url

  println(Url.parseOption("http://i❤️.ws").get.toStringPunycode) // This is fine
  println(Url.parseOption("http://👸🏽.cf").get.toStringPunycode) // Throws an exception
}
```
throws
```
http://xn--i-7iq.ws
Exception in thread "main" java.lang.ExceptionInInitializerError
	at onlinenslookup.analysis.ReproducePunycodeBug.main(ReproducePunycodeBug.scala)
Caused by: java.lang.IllegalArgumentException: java.text.ParseException: An unassigned code point was found in the input 👸🏽
	at java.base/java.net.IDN.toASCIIInternal(IDN.java:275)
	at java.base/java.net.IDN.toASCII(IDN.java:123)
	at java.base/java.net.IDN.toASCII(IDN.java:152)
	at io.lemonlabs.uri.inet.PunycodeSupport.toPunycode(PunycodeSupport.scala:7)
	at io.lemonlabs.uri.inet.PunycodeSupport.toPunycode$(PunycodeSupport.scala:5)
	at io.lemonlabs.uri.DomainName.toPunycode(Host.scala:114)
	at io.lemonlabs.uri.DomainName.toStringPunycode(Host.scala:180)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode$$anonfun$1(Uri.scala:857)
	at io.lemonlabs.uri.Authority.toString(Authority.scala:73)
	at io.lemonlabs.uri.AbsoluteUrl.toStringWithConfig(Uri.scala:1008)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode(Uri.scala:857)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode$(Uri.scala:721)
	at io.lemonlabs.uri.AbsoluteUrl.toStringPunycode(Uri.scala:966)
	at onlinenslookup.analysis.ReproducePunycodeBug$.<clinit>(ReproducePunycodeBug.scala:7)
	... 1 more
Caused by: java.text.ParseException: An unassigned code point was found in the input 👸🏽
	at java.base/jdk.internal.icu.text.StringPrep.map(StringPrep.java:302)
	at java.base/jdk.internal.icu.text.StringPrep.prepare(StringPrep.java:425)
	at java.base/java.net.IDN.toASCIIInternal(IDN.java:273)
	... 14 more
```

**Comment by theon** [comment]
Thanks for raising this! I will have a proper look in the next few days.

Currently scala-uri is using the java library support for IDN/punycode, so I think I'll start by trying some different JDK versions to see if support has been improved there at all. If not, maybe we'll need to look at moving to a better supported IDN/punycode library.

**Comment by theon** [comment]
StackOverflow [to the rescue](https://stackoverflow.com/questions/66048061/java-punycode-string). Fix is released in `4.0.1` and should be in maven central in a short while.

Thanks again for raising this.

**Comment by ruurtjan** [comment]
Awesome! Thanks for looking into it and for fixing it :)
